src/flb_plugin_proxy.c

```diff
@@ -183,37 +183,85 @@ static void flb_proxy_input_cb_resume(void *data, struct flb_config *config)
 }
 
 static void flb_plugin_proxy_destroy(struct flb_plugin_proxy *proxy);
-static int flb_proxy_output_cb_exit(void *data, struct flb_config *config)
+
+static int flb_proxy_output_cb_exit(void *out_context, struct flb_config *config)
 {
-    struct flb_output_plugin *instance = data;
-    struct flb_plugin_proxy *proxy = (instance->proxy);
+    struct flb_plugin_proxy_context *ctx = out_context;
+    struct flb_plugin_proxy *proxy = (ctx->proxy);
+
+    if (!out_context) {
+        return 0;
+    }
 
     if (proxy->def->proxy == FLB_PROXY_GOLANG) {
-        proxy_go_output_destroy(proxy->data);
+#ifdef FLB_HAVE_PROXY_GO
+        proxy_go_output_destroy(ctx);
+#endif
     }
-    flb_plugin_proxy_destroy(proxy);
+
+    flb_free(ctx);
     return 0;
 }
 
+static void flb_proxy_output_cb_destroy(struct flb_output_plugin *plugin)
+{
+    struct flb_plugin_proxy *proxy = (struct flb_plugin_proxy *) plugin->proxy;
+    /* cleanup */
+    void (*cb_unregister)(struct flb_plugin_proxy_def *def);
+
+    cb_unregister = flb_plugin_proxy_symbol(proxy, "FLBPluginUnregister");
+    if (cb_unregister != NULL) {
+        cb_unregister(proxy->def);
+    }
+
+    if (proxy->def->proxy == FLB_PROXY_GOLANG) {
+#ifdef FLB_HAVE_PROXY_GO
+        proxy_go_output_unregister(proxy->data);
+#endif
+    }
+
+    flb_plugin_proxy_destroy(proxy);
+}
+
 static int flb_proxy_input_cb_exit(void *in_context, struct flb_config *config)
 {
     struct flb_plugin_input_proxy_context *ctx = in_context;
+    struct flb_plugin_proxy *proxy = (ctx->proxy);
 
     if (!in_context) {
         return 0;
     }
 
-    if (ctx->proxy->def->proxy == FLB_PROXY_GOLANG) {
-        proxy_go_input_destroy(ctx->proxy->data);
+    if (proxy->def->proxy == FLB_PROXY_GOLANG) {
+#ifdef FLB_HAVE_PROXY_GO
+        proxy_go_output_destroy(ctx);
```

Comment:
Is proxy_go_input_destroy correct?

`proxy_go_output_destroy` does not proceed to handle `struct flb_plugin_input_proxy_context ` variable.

Reply:
You're right, I fixed it with next commit. I admit I haven't tested the input plugin, because the supporting code was not merged in fluent-bit-go repository. I'll try it now.